File: src/main/resources/public/src/app/service/car.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpParams} from "@angular/common/http";
import {Observable} from "rxjs/Observable";
import {DateService} from "../shared/services/date.service";

@Injectable()
export class CarService {

  constructor(private httpClient: HttpClient) {
  }

  saveCar(car: any): Observable<any>{
    let params = new HttpParams()
      .set('car', car);
    return this.httpClient.post('/api/saveCar',params);
  }

  getCar(carid: any): Observable<any> {
    return this.httpClient.get('/api/car/' + carid);
  }

  getCars(form: any): Observable<any> {
    let str = form.value['firstdate'];
    // console.log(form.value);
    // console.log(str);
    if(str != ''){

    }
    let params = new HttpParams()
      .set('carnum', form.value['carnum'])
      .set('carbrand', form.value['carbrand'])
      .set('carset', form.value['carset'])
      .set('firstdate1', str==''?'':str.split(',')[0])
      .set('firstdate2', str==''?'':str.split(',')[1])
      .set('creator', form.value['creator']);
    return this.httpClient.post('/api/cars', params);
  }

  getBrand(): Observable<any> {
    return this.httpClient.get('/api/brand').map(
      res=>{
        return res;
      },
      err=>{
        return err;
      }
    );
  }

  getSubBrand(brand: any): Observable<any> {
    let param = new HttpParams().set('brand', brand);
    return this.httpClient.post('/api/subBrand', param);
  }

}

export class Car {
  constructor(public carid: number,
              public carnum: string,
              public carbrand: string,
              public carset: string,
              public firstdate: string,
              public createdate: string,
              public creator: string,
              public add1: string,
              public add2: string,
              public add3: string) {
  }
}

File: src/main/java/com/joshua/easypass/service/CarService.java
package com.joshua.easypass.service;

import com.joshua.easypass.entity.Car;
import com.joshua.easypass.repository.CarRepository;
import com.joshua.easypass.repository.CarinfoRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Date;
import java.util.List;

@Service
public class CarService {
    public final static Logger logger = LoggerFactory.getLogger(CarService.class);
    @Autowired
    private CarRepository carRepo;
    @Autowired
    private CarinfoRepository carinfoRepo;

    public List<String> getBrand(){
        return carinfoRepo.getBrand();
    }

    public List<String> getSubBrand(String brand){
        return carinfoRepo.getSet(brand);
    }

    public Car[] getAllCars(){
        return carRepo.getAllCars();
    }

    public Car[] getCars(String carnum, String carbrand, String carset, Date firstdate1, Date firstdate2, String creator ){
        return carRepo.getCars(carnum,carbrand,carset,firstdate1,firstdate2,creator);
    }

    public void saveCar(Car car){
        carRepo.save(car);
    }

    public Car getCar(Integer carid){
        return carRepo.getCar(carid);
    }

}
